### PR TITLE
[CMake] Copy localization files to `share/diagnostics`

### PR DESCRIPTION
**Pull request by HassanElDesouky**
In this PR, I'm copying the YAML localization files to `share/diagnostics` regardless of whether it’s in toolchain or regular build. That way we won't need to use `-localization-path` flag each time.

The approach I'm taking with copying files with CMake *might* fail because of the limitation on Windows on the number of characters in a single command line, which is either 2047 or 8191 characters (as appropriate to the operating system). See [Microsoft's command-line string limitation](https://support.microsoft.com/en-gb/help/830473/command-prompt-cmd-exe-command-line-string-limitation). However, I don't think this will be an issue for now because the number of languages in [LocalizationLanguages](https://github.com/apple/swift/blob/master/include/swift/AST/LocalizationLanguages.def) is 184 languages and each code has a length of 2 characters and the extension of the localization files is `.yaml` so adding all of these up will be just 7 characters, multiplied by 184 is equal to 1288. Therefore, *I think* we will be safe with this method. WDYT?

cc @xedin, @owenv, @compnerd

### CODE REVIEW

**Comment by xedin** [comment]
@compnerd Does it make sense to you?

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test

File: CMakeLists.txt [hunk 1]
```diff
@@ -1113,6 +1113,12 @@ add_subdirectory(utils)
 
 add_subdirectory(userdocs)
 
+file(GLOB LOCALIZATION_FILES
+  "include/swift/AST/diagnostics/*.yaml"
+)
+
+file(COPY ${LOCALIZATION_FILES} DESTINATION "share/diagnostics")
```
Comment (xedin) [inline CMakeLists.txt:1120]: I think we might want to do something like:

```
file(GLOB LOCALIZATION_FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/include/swift/AST/diagnostics/*.yaml)

if (SWIFT_BUILT_STANDALONE)
  file(COPY ${LOCALIZATION_FILES} DESTINATION "share/swift/diagnostics")
else()
  swift_install_in_component(FILES ${LOCALIZATION_FILES}
                             DESTINATION "share/swift/diagnostics"
                             COMPONENT compiler)
endif()
```

But I'm not a CMake expert so any input is welcome. /cc @shahmishal

Comment (compnerd) [reply CMakeLists.txt:1120]: Globs are generally to be avoided in CMake.  The glob executes once and changes will _not_ be detected.  It really is better to explicitly enumerate the files.

Comment (HassanElDesouky) [reply CMakeLists.txt:1120]: What is the recommended way to copy files then?

**Pull request closed without merging**